pkg/action/executor/move_util.go

```diff
@@ -451,8 +451,9 @@ func getPodOwnersInfo(clusterScraper *cluster.ClusterScraper, pod *api.Pod,
 	return parent, nil, nsParentClient, nil, "", nil
 }
 
-// resourceRollout pauses/unpauses the rollout of a deployment or a deploymentconfig
-func resourceRollout(client dynamic.ResourceInterface, obj *unstructured.Unstructured, pause bool) error {
+// ResourceRollout pauses/unpauses the rollout of a deployment or a deploymentconfig
+// This additionally adds or removes the garbage collection label on the resource
+func ResourceRollout(client dynamic.ResourceInterface, obj *unstructured.Unstructured, pause bool) error {
```

Comment:
Similarly, let's split it into pauseRollout and resumeRollout. Using a boolean here maybe more error prone.

Reply:
I have a different opinion:
Its a lot of duplication of code and what we get in terms of the clarity is the `Pause` in the name of the function call in place of the name of argument. For example this is the implementation of `PauseRollout` (Which will be exact same implementation for `UnpauseRollout` except the flag value `true` in `PauseRollout`).
```
func PauseRollout(client dynamic.ResourceInterface, obj *unstructured.Unstructured) error {
	kind := obj.GetKind()
	name := obj.GetName()
	// This takes care of conflicting updates for example by operator
	// Ref: https://github.com/kubernetes/client-go/blob/master/examples/dynamic-create-update-delete-deployment/main.go
	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
		objCopy, err := client.Get(context.TODO(), name, metav1.GetOptions{})
		if err != nil {
			return err
		}

		if err := unstructured.SetNestedField(objCopy.Object, true, "spec", "paused"); err != nil {
			return err
		}
		addGCLabel(objCopy)

		_, err = client.Update(context.TODO(), objCopy, metav1.UpdateOptions{})
		if err != nil {
			return err
		}
		return nil
	})

	if retryErr != nil {
		return fmt.Errorf("error setting 'spec.paused' to true for %s %s: %v", kind, name, retryErr)
	}
	return nil
}
```

To ensure clarity in such cases, I always try to use self documenting code  as below:
```
			pause := true
			unpause := false
```
and then use these names while passing the arguments.